src/seedu/addressbook/data/person/Address.java

```diff
@@ -6,26 +6,23 @@
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
-public class Address {
+public class Address extends Contact{
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
-    private boolean isPrivate;
 
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
+        super(address, isPrivate);
```

Comment:
As discussed in class, you can abstract this even more by doing validation in the super class. See [this](https://github.com/nus-cs2103-AY1617S1/addressbook-level2/pull/1064/commits/18db3493b65da8e1a7be4fc54ebeff0e2ccf50b9)